Cabal/Distribution/Version.hs

```diff
@@ -105,13 +107,23 @@ import Control.Exception (assert)
 -- 'Binary' instance using a different (and more compact) encoding.
 --
 -- @since 2.0
-data Version = Version [Int]
-             deriving (Data,Eq,Ord,Generic,Show,Read,Typeable)
+
+data Version = PV0 {-# UNPACK #-} !Word64
```

Comment:
In particular mention the invariant, since that explains why we can use a derived Eq (there's a canonical representation)